platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkUtils.java

```diff
@@ -90,8 +93,12 @@ public static void handleNetworkException(final Exception e, final Connection co
         // we use a different marker depending on what the root cause is
         final Marker marker = NetworkUtils.determineExceptionMarker(e);
         if (SOCKET_EXCEPTIONS.getMarker().equals(marker)) {
-            final String formattedException = NetworkUtils.formatException(e);
-            logger.warn(marker, "Connection broken: {} {}", description, formattedException);
+            if (socketExceptionRateLimiter.requestAndTrigger()) {
+                logger.warn(marker, "Connection broken: {}", description, e);
+            } else {
+                final String formattedException = NetworkUtils.formatException(e);
+                logger.warn(marker, "Connection broken: {} {}", description, formattedException);
+            }
         } else {
             logger.error(EXCEPTION.getMarker(), "Connection broken: {}", description, e);
```

Comment:
same thing here though this isn't part of your change. If it is, could you pls fix?

Reply:
Please see above.